### DISCUSSION TITLE
Extended Sequence Number (ESN) support in IPSEC

### DISCUSSION THREAD

**Discussion by pankajbansal3073** [strongswan/strongswan#820] [Ideas]
Hi,

is there any plan to support Extended Sequence Number (ESN) support as mentioned in [RFC ](https://datatracker.ietf.org/doc/html/rfc4303#section-2.2.1) in IPSEC implementation ? i see that only [32 bit sequence number](https://github.com/strongswan/strongswan/blob/master/src/libipsec/esp_packet.c#L286) is inserted in encrypted packet as of now.

**Comment by tobiasbrunner** [comment]
There are currently no plans to do so as there really doesn't seem to be any need for it (_libipsec_ is not intended for high volume/performance use, see [kernel-libipsec](https://wiki.strongswan.org/projects/strongswan/wiki/kernel-libipsec)).

**Reply by pankajbansal3073** [reply]
Thanks for prompt reply.

As per RFC "Use of an ESN MUST be negotiated by an SA management protocol.  Note that in IKEv2, this negotiation is implicit; the default is ESN unless 32-bit sequence numbers are explicitly negotiated."

Does that mean ipsec communication fail with libipsec and default IKEv2 ? does that mean if libipsec is to be used then 32-bit sequence numbers MUST explicitly be negotiated in IKEv2 ?

**Reply by tobiasbrunner** [reply]
> Does that mean ipsec communication fail with libipsec and default IKEv2 ?

strongSwan's default is to add `noesn` to ESP/AH proposals. Users have to explicitly add `esn` to a configured proposal to propose the use ESN to the peer. Doing that wouldn't work with _libipsec_ (installing the IPsec SA will fail), so just make sure none of the configs specify `esn` in their proposals.
